src/base/misc/fatfs.c

```diff
@@ -1527,6 +1527,7 @@ void mimic_boot_blk(void)
 
     case MIDDRD_D:		/* DR-DOS with IBM compatibility naming */
       LWORD(edx) = f->drive_num;
+      LWORD(ebp) = LWORD(esp) = 0x7c00;
```

Comment:
It's inferred (but not very well) by the fact that the straight conversion 32 bit loader patch broke DR-DOS 5 & 6 floppy directory boot. Quite why it didn't harm hd directory boot I'm not sure. Anyway in the old fatfs_boot.S code the stack was set up for its own purposes, now  loading is done in 32 land, the stack wasn't set to a sane place (0x0000) I seem to remember.
As a note, these four patches fix nothing additional (except perhaps the MS-DOS 7 boot message display), it's just the conversion to 32 loader, then fix what got broken.